frb_example/pure_dart/dart/lib/main.dart

```diff
@@ -966,6 +977,73 @@ void main(List<String> args) async {
       expect(data2.isStale(), isFalse);
     });
   });
+
+  group('extended sync', () {
+    test('create', () {
+      var data = api.syncCreateOpaque();
+      data.dispose();
+    });
+
+    test('double call', () {
+      var data = api.syncCreateSyncOpaque();
+      expect(
+          api.syncRunOpaque(opaque: data),
+          "content - Some(PrivateData "
+          "{"
+          " content: \"content nested\", "
+          "primitive: 424242, "
+          "array: [451, 451, 451, 451, 451, 451, 451, 451, 451, 451], "
+          "lifetime: \"static str\" "
+          "})");
+      expect(
+          api.syncRunOpaque(opaque: data),
+          "content - Some(PrivateData "
+          "{"
+          " content: \"content nested\", "
+          "primitive: 424242, "
+          "array: [451, 451, 451, 451, 451, 451, 451, 451, 451, 451], "
+          "lifetime: \"static str\" "
+          "})");
+      data.dispose();
+    });
+
+    test('call after drop', () {
+      var data = api.syncCreateSyncOpaque();
+      expect(
+          api.syncRunOpaque(opaque: data),
+          "content - Some(PrivateData "
+          "{"
+          " content: \"content nested\", "
+          "primitive: 424242, "
+          "array: [451, 451, 451, 451, 451, 451, 451, 451, 451, 451], "
+          "lifetime: \"static str\" "
+          "})");
+      data.dispose();
+      expect(() => api.syncRunOpaque(opaque: data), throwsA(isA<FfiException>()));
+    });
+
+    test('option', () async {
+      var data = api.syncOption();
+      var data2 = api.syncOptionNull();
+      var data3 = api.syncOptionRustOpaque();
+      var data4 = api.syncOptionDartOpaque(opaque: () => () => 'magic');
+      expect(data, isNotNull);
+      expect(data2, isNull);
+      expect(data3, isNotNull);
+      expect(data4, isNotNull);
+      data3?.dispose();
```

Comment:
uber-nit: if it is not null, use `!`, since we are in tests and want to expose as many possible bugs as possible